File: CppPractice/CppLeetcode/Leetcode69.cpp
class Solution {
public:
    int mySqrt(int x) 
    {
        if(x==0){return 0;}
        if(x==1){return 1;}
        long long right = x;
        long long left = 1 ;   
        while(left<right)
        {
            long long mid= (right+left)/2;
            if(mid*mid>x){right=mid-1;}
            else if(mid*mid==x){return mid;}
            else{left=mid+1;}
        }
        if(left*left>x){return left-1;}
        else{return left;}

    }
};

File: CSharpPractice/LeetcodePractice/Leetcode69.cs
public class Solution69 {
    public int MySqrt(int x) 
    {
        long left = 1;
        long right = x/2;
        if(x==0 || x==1){return x;}
        long mid = 0;
        long square =0;
        while(left<=right)
        {
            mid = (left+right)/2;
            square = mid*mid;
            if(square<x)
            {
                left = mid+1;
            }
            else if(square>x)
            {
                right = mid-1;
            }
            else{return (int)mid;}
        }
        Console.WriteLine(right);
        if((right+1) * (right+1)<=x){return (int)right+1;}
        else if(right*right>x){return (int)right-1;}
        else{return (int)right;}
        
    }
}